lib/bios/menus.robot

```diff
@@ -410,11 +410,23 @@ Reset To Defaults Tianocore
     [Documentation]    Resets all Tianocore options to defaults. It is invoked
     ...    by pressing F9 and confirming with 'y' when in option
     ...    setting menu.
+    Power On
+    ${main_menu}=    Enter Setup Menu Tianocore And Return Construction
     Read From Terminal
     Press Key N Times    1    ${F9}
     Read From Terminal Until    ignore.
     Write Bare Into Terminal    y
 
+    IF    ${DASHARO_SERIAL_PORT_MENU_SUPPORT}
+        ${dasharo_menu}=    Enter Dasharo System Features    ${main_menu}
+        ${serial_menu}=    Enter Dasharo Submenu    ${dasharo_menu}    Serial Port Configuration
+        TRY
+            Set Option State    ${serial_menu}    Enable COM0 Serial    ${TRUE}
```

Comment:
Why is it needed? Can we have different naming here on different platforms? This deserves a comment at least.

Reply:
https://github.com/Dasharo/open-source-firmware-validation/pull/601/commits/4364e09f38fab8ea9bca2f303d4b42a33af0280b

Reply:
> # The Serial Redirection option is called differently in some versions of Dasharo

This could be more specific - which platform version you have spotted this on.

This is q bit unfortunate workaround, as we do not fix the problem at it's roots, we only workaround one of the instances. If we use this menu elsewhere, we will hit this problem again.

> maybe the menus.py library could be modified a bit?

Definitely, we should fix (or apply a workaround) there. Do you have some ideas?

As a last restort, we can create issue for this, but ideally we at least apply workaround (with a proper comments) in the library, not in one of it's calls.

Reply:
`Enable COM0 Serial Console Redirection`  <- we changed it per Protectli request at some point.

Reply:
This library is actually only used in the `Set/Get UEFI Option` keywords and is not used in this test case. The way the Serial Redirection is set here is a standard among other tests, for example the `reset-to-defaults.robot`. The paths are hard-coded into the tests and if the path changes in some version of Dasharo, all the tests that use it would need to be updated to check or alternatives. I think this is a reasonable workaround for now until we want to do a rework of how the options are being found in all of the tests.

And about the possible rework:
The main question here is how much of variance in the paths is accepted in the future? Will it only be alternative names of single options? What about the names of submenus? Maybe whole paths will be changed, or the options will be moved between existing menus?

As for my ideas for the rework:
If the `menus.py` library was used everywhere menus are traversed then exceptions in names could be easy to fix. I'd imagine the library could just keep alternative paths to the options and the alternatives could be tried one by one.

So instead of a list of steps:
```
    "SerialRedirection": [
        "Dasharo System Features",
        "Serial Port Configuration",
        "Enable Serial Port Console Redirection",  # Enable COM0 Serial Console Redirection??
    ],
```
A tree could be used. 
```
    "SerialRedirection": [
        "Dasharo System Features": [
            "Serial Port Configuration": [
                "Enable Serial Port Console Redirection",
                "Enable COM0 Serial Console Redirection"
            ]
        ]
    ],
```
This would easily solve simple renames, but if the option would be moved between two existing menus then finding it could require traversing the menus backwards ...
```
     "A": [
        "B": [ # B exists, enter
            "Option" # No "Option" in this submenu
        ],
        "C": [ # Need to go up to try "C"
            "Option" # "Option" found
        ]
    ],
```

Also there's the option to define paths and menu option names in the menus as config variables, which would need creating and maintaining a whole lot of them. There are 15 paths, 7 submenus and 16 option names defined in menus.py. And there already are too many variables in the configs to remember. 
```
${SETUP_SUBMENU_DASHARO_SYSTEM_FEATURES}=    "Dasharo System Features"
${SETUP_SUBMENU_SERIAL_PORT_CONFIG}=    "Serial Port Configuration"
${SETUP_OPTION_SERIAL_PORT_REDIRECTION}=    "Enable COM0 Serial Console Redirection"
@{OPTION_PATH_SERIAL_PORT_REDIRECTION}=
...    ${SETUP_SUBMENU_DASHARO_SYSTEM_FEATURES}
...    ${SETUP_SUBMENU_SERIAL_PORT_CONFIG}
...    ${SETUP_OPTION_SERIAL_PORT_REDIRECTION}
```
This way though any future change in the menus structure would be easily fixable on per-platform basis and there would be no need to dynamically look around for where the options are located, which could take a lot of time depending on the amount of possible alternative paths.

Reply:
We definitely would like to keep the menu names in a single place. If Dasharo versioning would be standardized, we could define which menu layout corresponds to the given Dasharo version. Let's move your ideas to an issue so we can continue discussing there.

Reply:
@philipandag merging already, but please remembr  of this issue

Reply:
Ok, there it is: https://github.com/Dasharo/open-source-firmware-validation/issues/614